docs/core/compatibility/sdk/7.0/custom-serialization.md

```diff
@@ -0,0 +1,32 @@
+---
+title: "Breaking change: Serialization of custom types in .NET 7"
+description: Learn about a breaking change in MSBuild for .NET 7 where serialization using BinaryFormatter of certain user-defined types is removed.
+author: baronfel
+ms.date: 05/06/2022
+---
+
+# BinaryFormatter serialization of custom BuildEventArgs and ITaskItems removed for .NET 7
+
+MSBuild in .NET 7 doesn't support serialization of custom BuildEventArgs and ITaskItems via the BinaryFormatter serializer.
+
+## Version Introduced
+
+MSBuild 17.4 (.NET SDK 7.0.100)
+
+## Old Behavior
+
+MSBuild used BinaryFormatter via the `TranslateDotNet` method to translate custom BuildEventArgs and ITaskItems that users could define in their own Tasks.
```

Comment:
`TranslateDotNet` is internal and no one should know or care about it.

```suggestion
MSBuild used BinaryFormatter to preserve custom types across certain boundaries, most notably when running in a multi-process environment.
```

That's awful vague though.

Reply:
MSBuild used BinaryFormatter to serialize and deserialize custom types derived from BuildEventArgs and ITaskItem, especially between processes when building with multiple processes.
?